apps/src/templates/rubrics/teacherRubricRedux.ts

```diff
@@ -0,0 +1,88 @@
+import {PayloadAction, createSlice, createAsyncThunk} from '@reduxjs/toolkit';
+
+interface TeacherEvaluation {
+  feedback: string;
+  id: number;
+  learning_goal_id: number;
+  understanding: number;
+}
+
+interface UserEvalData {
+  eval: TeacherEvaluation[];
+  user_family_name?: string;
+  user_id: number;
+  user_name: string;
+}
+
+type AllTeacherEvaluationData = UserEvalData[];
+
+interface HasTeacherFeedbackMap {
+  [key: number]: boolean;
+}
+
+interface TeacherRubricState {
+  allTeacherEvaluationData: AllTeacherEvaluationData;
+  hasTeacherFeedbackMap: HasTeacherFeedbackMap;
+}
+
+const initialState: TeacherRubricState = {
+  allTeacherEvaluationData: [],
+  hasTeacherFeedbackMap: {},
+};
+
+const teacherRubricReduxSlice = createSlice({
+  name: 'teacherRubric',
+  initialState,
+  reducers: {
+    setAllTeacherEvaluationData(
+      state,
+      action: PayloadAction<AllTeacherEvaluationData>
+    ) {
+      state.allTeacherEvaluationData = action.payload;
+    },
+    setHasTeacherFeedbackMap(
+      state,
+      action: PayloadAction<HasTeacherFeedbackMap>
+    ) {
+      state.hasTeacherFeedbackMap = action.payload;
+    },
+  },
+});
+
+export const loadAllTeacherEvaluationData = createAsyncThunk(
+  'teacherRubric/loadAllTeacherEvaluationData',
+  async (params: {rubricId: number; sectionId: number}, thunkAPI) => {
+    const {rubricId, sectionId} = params;
+
+    const fetchTeacherEvaluationAll = (rubricId: number, sectionId: number) => {
+      return fetch(
+        `/rubrics/${rubricId}/get_teacher_evaluations_for_all?section_id=${sectionId}`
+      );
+    };
+
+    const initializeHasTeacherFeedbackMap = (
+      allTeacherEvaluationData: AllTeacherEvaluationData
+    ) => {
+      const hasFeedbackMap: HasTeacherFeedbackMap = {};
+      allTeacherEvaluationData.forEach(userEvalData => {
+        if (userEvalData?.user_id) {
```

Comment:
The type declaration above asserts that a `UserEvalData` will have a defined numeric `user_id`. Either the type declaration is incorrect or this check is unnecessary.

Reply:
good catch. it looks like this should always be defined: https://github.com/code-dot-org/code-dot-org/blob/012614906ee011eff3f8da499e011848d15100f5/dashboard/app/controllers/rubrics_controller.rb#L130